app/services/wildcard_url_checker.rb

```diff
@@ -0,0 +1,10 @@
+module WildcardUrlChecker
+
+  def self.check_url(url, url_to_check)
+    escaped_url = Regexp.escape(url).sub("\\*", '\S*')
```

Comment:
I think we want `gsub` here to support "`*.example.com*`"

```suggestion
    escaped_url = Regexp.escape(url).gsub("\\*", '\S*')
```